numpy/linalg/lapack_lite/f2c.h

```diff
@@ -214,4 +214,171 @@ typedef doublereal E_f;	/* real function with -R not specified */
+extern integer i_len(char *, ftnlen);
+extern integer i_mod(integer *, integer *);
+extern integer i_nint(float *);
+extern integer i_sign(integer *, integer *);
+extern integer iargc_(void);
+extern ftnlen l_ge(char *, char *, ftnlen, ftnlen);
+extern ftnlen l_gt(char *, char *, ftnlen, ftnlen);
+extern ftnlen l_le(char *, char *, ftnlen, ftnlen);
+extern ftnlen l_lt(char *, char *, ftnlen, ftnlen);
+extern void pow_ci(complex *, complex *, integer *);
+extern double pow_dd(double *, double *);
+extern double pow_di(double *, integer *);
+extern short pow_hh(short *, shortint *);
+extern integer pow_ii(integer *, integer *);
+extern double pow_ri(float *, integer *);
+extern void pow_zi(doublecomplex *, doublecomplex *, integer *);
+extern void pow_zz(doublecomplex *, doublecomplex *, doublecomplex *);
+extern double r_abs(float *);
+extern double r_acos(float *);
+extern double r_asin(float *);
+extern double r_atan(float *);
+extern double r_atn2(float *, float *);
+extern void r_cnjg(complex *, complex *);
+extern double r_cos(float *);
+extern double r_cosh(float *);
+extern double r_dim(float *, float *);
+extern double r_exp(float *);
+extern float r_imag(complex *);
+extern double r_int(float *);
+extern float r_lg10(real *);
+extern double r_log(float *);
+extern double r_mod(float *, float *);
+extern double r_nint(float *);
+extern double r_sign(float *, float *);
+extern double r_sin(float *);
+extern double r_sinh(float *);
+extern double r_sqrt(float *);
+extern double r_tan(float *);
+extern double r_tanh(float *);
+extern void s_cat(char *, char **, integer *, integer *, ftnlen);
+extern integer s_cmp(char *, char *, ftnlen, ftnlen);
+extern void s_copy(char *, char *, ftnlen, ftnlen);
+extern int s_paus(char *, ftnlen);
+extern integer s_rdfe(cilist *);
+extern integer s_rdue(cilist *);
+extern integer s_rnge(char *, integer, char *, integer);
+extern integer s_rsfe(cilist *);
+extern integer s_rsfi(icilist *);
+extern integer s_rsle(cilist *);
+extern integer s_rsli(icilist *);
+extern integer s_rsne(cilist *);
+extern integer s_rsni(icilist *);
+extern integer s_rsue(cilist *);
+extern int s_stop(char *, ftnlen);
+extern integer s_wdfe(cilist *);
+extern integer s_wdue(cilist *);
+extern void s_wsfe(	cilist *);
+extern integer s_wsfi(icilist *);
+extern integer s_wsle(cilist *);
+extern integer s_wsli(icilist *);
+extern integer s_wsne(cilist *);
+extern integer s_wsni(icilist *);
+extern integer s_wsue(cilist *);
+extern void sig_die(char *, int);
+extern integer signal_(integer *, void (*)(int));
+extern integer system_(char *, ftnlen);
+extern double z_abs(doublecomplex *);
+extern void z_cos(doublecomplex *, doublecomplex *);
+extern void z_div(doublecomplex *, doublecomplex *, doublecomplex *);
+extern void z_exp(doublecomplex *, doublecomplex *);
+extern void z_log(doublecomplex *, doublecomplex *);
+extern void z_sin(doublecomplex *, doublecomplex *);
+extern void z_sqrt(doublecomplex *, doublecomplex *);
```

Comment:
I'm a little wary of the fact that either:
* You created this manually
* You created this with an automated process that isn't documented in a way that can be repeated in future.

Should we perhaps have the script check the list of headers in this file against the ones it removed from the source, and print out warnings/errors if some are missing?

Reply:
These headers come directly from `libf2c` (specifically, some random gh repository).

> Should we perhaps have the script check the list of headers in this file against the ones it removed from the source, and print out warnings/errors if some are missing?

The compiler can check this as long as there are no inline prototypes (which is why inline prototypes are a really bad idea). The only script that we need to write is one which will remove the inline prototypes, which is what I have done here.

Reply:
Can you link to that random repo? Does this include the `Subroutine` declarations too?

Reply:
I think it's [here](https://github.com/juanjosegarciaripoll/f2c/blob/master/include/f2c.h). But again, all that's required is to remove the inline prototypes and the compiler will check consistency. It's not required to trust that these definitions are correc.t

Reply:
> the compiler will check consistency. I

Is that true if one is missing? (hence my question)

The one you link to [at juanjosegarciaripoll/f2c](https://github.com/juanjosegarciaripoll/f2c/blob/master/include/f2c.h) doesn't match the [debian package](
https://anonscm.debian.org/cgit/collab-maint/f2c.git/tree/f2c.h)...

Reply:
> Is that true if one is missing? (hence my question)

Yes. There is only a problem when the prototypes are inconsistently defined across multiple files.